File: test/aidbox_sdk/snapshots/dotnet/hl7-fhir-r4-core/Condition.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class Condition : DomainResource, IResource
{
    public object? Onset    
    {
        get
        {
            if (OnsetRange is not null)
            {
                return OnsetRange;
            }
    
            if (OnsetAge is not null)
            {
                return OnsetAge;
            }
    
            if (OnsetPeriod is not null)
            {
                return OnsetPeriod;
            }
    
            if (OnsetString is not null)
            {
                return OnsetString;
            }
    
            if (OnsetDateTime is not null)
            {
                return OnsetDateTime;
            }
    
            return null;
        }
    
        set
        {
            if (value?.GetType() == typeof(Base.Range))
            {
                OnsetRange = (Base.Range)value;return;
            }
    
            if (value?.GetType() == typeof(string))
            {
                OnsetAge = (string)value;return;
            }
    
            if (value?.GetType() == typeof(Base.Period))
            {
                OnsetPeriod = (Base.Period)value;return;
            }
    
            if (value?.GetType() == typeof(string))
            {
                OnsetString = (string)value;return;
            }
    
            if (value?.GetType() == typeof(string))
            {
                OnsetDateTime = (string)value;return;
            }
    
            throw new ArgumentException("Invalid type provided");
        }
    }
    public Base.CodeableConcept[]? Category { get; set; }
    public Base.CodeableConcept? ClinicalStatus { get; set; }
    public string? AbatementAge { get; set; }
    public Base.Range? OnsetRange { get; set; }
    public string? OnsetAge { get; set; }
    public Condition_Stage[]? Stage { get; set; }
    public Base.ResourceReference? Encounter { get; set; }
    public Condition_Evidence[]? Evidence { get; set; }
    public Base.Period? OnsetPeriod { get; set; }
    public object? Abatement    
    {
        get
        {
            if (AbatementAge is not null)
            {
                return AbatementAge;
            }
    
            if (AbatementPeriod is not null)
            {
                return AbatementPeriod;
            }
    
            if (AbatementString is not null)
            {
                return AbatementString;
            }
    
            if (AbatementRange is not null)
            {
                return AbatementRange;
            }
    
            if (AbatementDateTime is not null)
            {
                return AbatementDateTime;
            }
    
            return null;
        }
    
        set
        {
            if (value?.GetType() == typeof(string))
            {
                AbatementAge = (string)value;return;
            }
    
            if (value?.GetType() == typeof(Base.Period))
            {
                AbatementPeriod = (Base.Period)value;return;
            }
    
            if (value?.GetType() == typeof(string))
            {
                AbatementString = (string)value;return;
            }
    
            if (value?.GetType() == typeof(Base.Range))
            {
                AbatementRange = (Base.Range)value;return;
            }
    
            if (value?.GetType() == typeof(string))
            {
                AbatementDateTime = (string)value;return;
            }
    
            throw new ArgumentException("Invalid type provided");
        }
    }
    public Base.Period? AbatementPeriod { get; set; }
    public Base.ResourceReference? Asserter { get; set; }
    public Base.Annotation[]? Note { get; set; }
    public string? AbatementString { get; set; }
    public Base.Range? AbatementRange { get; set; }
    public string? RecordedDate { get; set; }
    public string? OnsetString { get; set; }
    public Base.ResourceReference? Recorder { get; set; }
    public Base.CodeableConcept? Severity { get; set; }
    public Base.CodeableConcept? Code { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public string? OnsetDateTime { get; set; }
    public Base.CodeableConcept[]? BodySite { get; set; }
    public string? AbatementDateTime { get; set; }
    public Base.CodeableConcept? VerificationStatus { get; set; }
    public required Base.ResourceReference Subject { get; set; }

    public class Condition_Stage : BackboneElement
    {
        public Base.CodeableConcept? Type { get; set; }
        public Base.CodeableConcept? Summary { get; set; }
        public Base.ResourceReference[]? Assessment { get; set; }
    }

    public class Condition_Evidence : BackboneElement
    {
        public Base.CodeableConcept[]? Code { get; set; }
        public Base.ResourceReference[]? Detail { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/Condition.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Annotation
from base import Age
from base import Period
from base import CodeableConcept
from base import Range
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class ConditionStage(BackboneElement):
    type: Optional[CodeableConcept] = None
    summary: Optional[CodeableConcept] = None
    assessment: Optional[List[Reference]] = field(default_factory=list)

@dataclass(kw_only=True)
class ConditionEvidence(BackboneElement):
    code: Optional[List[CodeableConcept]] = field(default_factory=list)
    detail: Optional[List[Reference]] = field(default_factory=list)

@dataclass(kw_only=True)
class Condition(DomainResource):
    category: Optional[List[CodeableConcept]] = field(default_factory=list)
    clinical_status: Optional[CodeableConcept] = None
    abatement_age: Optional[Age] = None
    onset_range: Optional[Range] = None
    onset_age: Optional[Age] = None
    stage: Optional[List[ConditionStage]] = field(default_factory=list)
    encounter: Optional[Reference] = None
    evidence: Optional[List[ConditionEvidence]] = field(default_factory=list)
    onset_period: Optional[Period] = None
    abatement_period: Optional[Period] = None
    asserter: Optional[Reference] = None
    note: Optional[List[Annotation]] = field(default_factory=list)
    abatement_string: Optional[str] = None
    abatement_range: Optional[Range] = None
    recorded_date: Optional[str] = None
    onset_string: Optional[str] = None
    recorder: Optional[Reference] = None
    severity: Optional[CodeableConcept] = None
    code: Optional[CodeableConcept] = None
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    onset_date_time: Optional[str] = None
    body_site: Optional[List[CodeableConcept]] = field(default_factory=list)
    abatement_date_time: Optional[str] = None
    verification_status: Optional[CodeableConcept] = None
    subject: Reference